tests/lib/rules/sort-vars.js

```diff
@@ -34,8 +34,37 @@ eslintTester.addRuleTest("lib/rules/sort-vars", {
         { code: "var a, A;", args: ignoreCaseArgs },
         { code: "var A, a;", args: ignoreCaseArgs },
         { code: "var a, B, c;", args: ignoreCaseArgs },
-        { code: "var A, b, C;", args: ignoreCaseArgs }
-
+        { code: "var A, b, C;", args: ignoreCaseArgs },
+        { code: "var {a, b, c};", args: ignoreCaseArgs,
+            ecmaFeatures: { destructuring: true }},
+        { code: "var {A, b, C};", args: ignoreCaseArgs,
+            ecmaFeatures: { destructuring: true }},
+        { code: "var test = [1,2,3];", args: ignoreCaseArgs,
+            ecmaFeatures: { destructuring: true }},
+        { code: "var {a,b} = [1,2];", ecmaFeatures: { destructuring: true } },
+        { code: "var [a, B, c] = [1, 2, 3];", args: ignoreCaseArgs,
+            ecmaFeatures: { destructuring: true }},
+        { code: "var [A, B, c] = [1, 2, 3];", args: ignoreCaseArgs,
+            ecmaFeatures: { destructuring: true }},
+        { code: "var [A, b, C] = [1, 2, 3];", args: ignoreCaseArgs,
+            ecmaFeatures: { destructuring: true }},
+        { code: "let {a, b, c};", args: ignoreCaseArgs,
+            ecmaFeatures: { blockBindings: true, destructuring: true}},
+        { code: "let [a, b, c] = [1, 2, 3];", args: ignoreCaseArgs,
+            ecmaFeatures: { blockBindings: true, destructuring: true }},
+        { code: "const {a, b, c} = {a: 1, b: true, c: \"Moo\"};", args: ignoreCaseArgs,
+            ecmaFeatures: { blockBindings: true, destructuring: true }},
+        { code: "const [a, b, c] = [1, true, \"Moo\"];", args: ignoreCaseArgs,
+            ecmaFeatures: { blockBindings: true, destructuring: true }},
+        { code: "var {a, x: {b, c}} = {};", args: ignoreCaseArgs,
+            ecmaFeatures: { destructuring: true }},
+        { code: "var {a, x: [b, c]} = {};", args: ignoreCaseArgs,
+            ecmaFeatures: { destructuring: true }},
+        { code: "var [a, {b, c}] = {};", args: ignoreCaseArgs,
+            ecmaFeatures: { destructuring: true }},
+        { code: "var [a, {x: {b, c}}] = {};", args: ignoreCaseArgs,
+            ecmaFeatures: { destructuring: true }},
+        { code: "var a = 42, {b, c } = {};", args: ignoreCaseArgs, ecmaFeatures: { destructuring: true }}
```

Comment:
Right now all of the tests include sorted variables, even within destructuring. Can you include some valid tests with unsorted identifiers inside destructuring patterns?